HHS/Head-Start-TTADP
# CI: Static Security Scan, Code coverage (#37)

**Description of change**
Adds two features to the continuous integration pipeline:
- **static security scan via `yarn audit`** - The ternary operator is necessary because `yarn audit` returns a non-zero exit code if issues of any severity level are found. We only want to fail the job though if moderate or higher level vulnerabilities are found. Moderate vulnerabilities return an exit code of 4. More info on this is available in the [yarn docs](https://classic.yarnpkg.com/en/docs/cli/audit/).
- **code coverage threshold** - Adds BE coverage threshold of 60, and FE coverage threshold of 90.  The BE is set at 60 simply because this is the highest threshold that will pass. I think we should add a ticket to the backlog to write tests to increase the coverage to 70%-80%. Writing new tests to increase testing coverage is outside of the scope of this ticket.

**How to test**
- Checkout out main branch in circle ci to see integration and deployment passing https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP?branch=main
- For an example of how the pipeline will fail when moderate vulnerabilities are found see https://app.circleci.com/pipelines/github/adhocteam/Head-Start-TTADP/304/workflows/5ad792fd-4c87-4a94-9596-2b1c14a05925 

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/11

**Checklist**

- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] Documentation updated - **_This issue requires an ADR that will be completed in a future PR that for this ticket_**
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified
